model/postgres/postgres_server.rb

```diff
@@ -13,7 +13,7 @@ class PostgresServer < Sequel::Model
   include Authorization::HyperTagMethods
   include Authorization::TaggableMethods
 
-  semaphore :restart, :destroy
+  semaphore :initial_provisioning, :restart, :destroy
```

Comment:
I feel like introducing a semaphore to be used only once is not really inline with the idea of semaphores. Maybe we can do it with a `state` column on the postgres model?

Reply:
I believe adding `state` to our models is not entirely consistent with our current model too. I prefer the semaphore approach. In Clover's processor (a.k.a Marlin), each runnable model includes columns to handle state machine runs, such as `state`, `deadline`, etc. In Clover, it seems we delegated the part that handles runs to `Strand`. (Another example is that we delegated columns related to ssh to `Sshable`). @fdr might provide further insight on this topic. The resource nexuses have labels to maintain the current state. There is a certain relationship between a model's state and a nexus's label. I think we should refrain from adding states and strive to keep them synchronized.

Reply:
I agree that adding `state` to the model is also not perfect but having a semaphore to be used a single time? Maybe another idea is to couple configure + restart into single state and create billing records if there is not one?

Reply:
"Second" state seems like a second source of truth, which I would like to avoid.

One benefit of the semaphore is that it would be deleted once decremented and there would be no trace left of it, which seems like a nice fit for one time use. `state` column is much more persistent.

I didn't put configure and restart into single label, because I plan to use them as individual labels in the future (when user triggers restart or when we want to re-run configure for example).

Reply:
BTW, destroy is also one time use semaphore :slightly_smiling_face:

Reply:
I leave the decision to you but this doesn't feel right to me 100%. destroy is okay because once we use it, entity is also gone. So, the same entity can only execute that code path once and for all.

Reply:
I'll merge it as is then. Here is my thinking. The new semaphore has 2 obvious costs;
- **Taking space in database:** I think this is also OK as it is quite short lived. There will be no trace of the semaphore after it is decremented.
- **Increasing overall maintenance cost despite being used one time:** I think this is OK because it is equivalent to destroy semaphore. Entity being around isn't too different from semaphores' perspective.

Also, alternative ways to achieve same thing seems to add more complexity. So I think this change is fine.